### PR TITLE
linux-kernel: add instructions about Xenomai4

### PR DESCRIPTION
**Pull request by ldts**
Add instructions about Xenomai4

## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by ldts** [comment]
> Docs for [534ceea](https://github.com/foundriesio/docs/commit/534ceeabb5802736c9d2c9d9dfb99237f8f20263) are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2201/docs/artifacts/html/index.html

Docs for [534ceea](https://github.com/foundriesio/docs/commit/534ceeabb5802736c9d2c9d9dfb99237f8f20263) are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2201/docs/artifacts/html/index.html

File: source/reference-manual/linux/linux-kernel.rst [hunk 1]
```diff
@@ -76,6 +76,38 @@ in ``meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc``::
...
+
+Building Linux microPlatform with linux-lmp-fslc-imx-xeno4
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Set ``PREFERRED_PROVIDER_virtual/kernel`` to
+``linux-lmp-fslc-imx-xeno4`` and ``MACHINE_FEATURES:append`` to ``xeno4``
+in ``meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc``::
+
+    $ cat meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc
+    PREFERRED_PROVIDER_virtual/kernel:mx8mm-nxp-bsp =
```
Comment (quaresmajose) [inline source/reference-manual/linux/linux-kernel.rst:106]: There is a extra new line in the PREFERRED_PROVIDER_virtual/kernel
```
    $ cat meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc
    PREFERRED_PROVIDER_virtual/kernel:mx8mm-nxp-bsp = "linux-lmp-fslc-imx-xeno4"
    MACHINE_FEATURES:append = " xeno4"
```

File: source/reference-manual/linux/linux-kernel.rst [hunk 2]
```diff
@@ -76,6 +76,38 @@ in ``meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc``::
...
+can be used to enable the Xenomai4 co-kernel on iMX boards.
+
+Like its predecessors in the Xenomai core series, `Xenomai4`_ with the
+`EVL core`_ brings real-time capabilities to Linux by embedding a
+companion core into the kernel, which specifically deals with tasks
+requiring ultra low and bounded response time to events.
+
+In this model, the general purpose kernel and the real-time core operate
+almost asynchronously, both serving their own set of tasks, always
+giving the latter precedence over the former.
```
Comment (kprosise) [inline source/reference-manual/linux/linux-kernel.rst:93]:
```suggestion
	Like its predecessors in the Xenomai core series, `Xenomai4`_ with the
	`EVL core`_ brings real-time capabilities to Linux by embedding a
	companion core into the kernel, which specifically deals with tasks
	requiring ultra low and bounded response time to events.

	In this model, the general purpose kernel and the real-time core operate
	almost asynchronously, both serving their own set of tasks, always
	giving the latter precedence over the former.
```

Block quote

**Review by kprosise — APPROVED** [review]
Minor suggestion provided

**Comment by doanac** [comment]
Docs for 3ee93d85001e903ebaa686eab55b2b897a473aff are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2216/docs/artifacts/html/index.html

**Comment by ldts** [comment]
suggestions applied

**Comment by doanac** [comment]
Docs for 1493deb4cf0f4c666723c34d1486bcbff79391cc are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2217/docs/artifacts/html/index.html

**Pull request merged**